src/Control/Email/Email.php

```diff
@@ -31,738 +16,646 @@ class Email extends ViewableData
     }
 
     /**
-     * Used by {@link SSViewer} templates to detect if we're rendering an email template rather than a page template
+     * @return int
      */
-    public function IsEmail()
+    public function getPriority()
     {
-        return true;
+        return $this->getSwiftMessage()->getPriority();
     }
 
     /**
-     * Populate this email template with values. This may be called many times.
-     *
-     * @param array|ViewableData $data
+     * @param int $priority
      * @return $this
      */
-    public function populateTemplate($data)
+    public function setPriority($priority)
     {
-        if ($this->template_data) {
-            $this->template_data = $this->template_data->customise($data);
-        } else {
-            if (is_array($data)) {
-                $data = new ArrayData($data);
-            }
-            $this->template_data = $this->customise($data);
-        }
-        $this->parseVariables_done = false;
+        $this->getSwiftMessage()->setPriority($priority);
 
         return $this;
     }
 
     /**
-     * Load all the template variables into the internal variables, including
-     * the template into body.  Called before send() or debugSend()
-     * $isPlain=true will cause the template to be ignored, otherwise the GenericEmail template will be used
-     * and it won't be plain email :)
-     *
-     * @param bool $isPlain
+     * @param string $path Path to file
      * @return $this
      */
-    protected function parseVariables($isPlain = false)
-    {
-        $origState = SSViewer::config()->get('source_file_comments');
-        SSViewer::config()->update('source_file_comments', false);
-
-        if (!$this->parseVariables_done) {
-            $this->parseVariables_done = true;
-
-            // Parse $ variables in the base parameters
-            $this->templateData();
-
-            // Process a .SS template file
-            $fullBody = $this->body;
-            if ($this->ss_template && !$isPlain) {
-                // Requery data so that updated versions of To, From, Subject, etc are included
-                $data = $this->templateData();
-                $candidateTemplates = [
-                    $this->ss_template,
-                    [ 'type' => 'email', $this->ss_template ]
-                ];
-                $template = new SSViewer($candidateTemplates);
-                if ($template->exists()) {
-                    $fullBody = $template->process($data);
-                }
-            }
+    public function addAttachment($path, $alias = null, $mime = null)
+    {
+        $attachment = \Swift_Attachment::fromPath($path);
+        if ($alias) {
+            $attachment->setFilename($alias);
+        }
+        if ($mime) {
+            $attachment->setContentType($mime);
+        }
+        $this->getSwiftMessage()->attach($attachment);
+
+        return $this;
+    }
 
-            // Rewrite relative URLs
-            $this->body = HTTP::absoluteURLs($fullBody);
```

Comment:
Actually, I've moved it to `setBody` so we always do this on the body regardless of render. However, maybe the `Mailer` or even `SwiftPlugin` should do it to absolutely make sure nothing can get through?